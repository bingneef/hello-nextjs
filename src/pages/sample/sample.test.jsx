import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Sample from "./index.tsx";

describe("Sample", () => {
  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Sample />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const component = renderer.create(<Sample title="Sample" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const rootInstance = component.root;
    const $button = rootInstance.findByType("button");
    expect($button.props.children).toBe("Secondary");
  });
});
