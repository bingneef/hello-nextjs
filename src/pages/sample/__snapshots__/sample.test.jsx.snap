// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample matches the snapshot 1`] = `
Array [
  <div
    className=""
  >
    Sample
  </div>,
  <div>
    0
  </div>,
  <button
    onClick={[Function]}
  >
    +
  </button>,
  <button
    onClick={[Function]}
  >
    -
  </button>,
  <button
    onClick={[Function]}
  >
    reset
  </button>,
  <button
    className=""
    data-baseweb="button"
    disabled={false}
    onClick={[Function]}
  >
    Secondary
  </button>,
]
`;
